<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="maximumSize">
       <size>
        <width>250</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="tabPosition">
       <enum>QTabWidget::West</enum>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tabCharges">
       <attribute name="title">
        <string>Заряды</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="gbCharge">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="title">
           <string>Выделение</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <property name="fieldGrowthPolicy">
            <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
           </property>
           <item row="1" column="0">
            <widget class="QLabel" name="lbX">
             <property name="text">
              <string>X</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="sbX">
             <property name="accelerated">
              <bool>true</bool>
             </property>
             <property name="maximum">
              <double>400.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lbY">
             <property name="text">
              <string>Y</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="sbY">
             <property name="accelerated">
              <bool>true</bool>
             </property>
             <property name="maximum">
              <double>400.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="lbCharge">
             <property name="text">
              <string>Заряд (Кл)</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="sbCharge">
             <property name="accelerated">
              <bool>true</bool>
             </property>
             <property name="decimals">
              <number>6</number>
             </property>
             <property name="minimum">
              <double>-100000.000000000000000</double>
             </property>
             <property name="maximum">
              <double>100000.000000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QPushButton" name="pbRemove">
             <property name="text">
              <string>Удалить</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="cbChargesVisible">
          <property name="text">
           <string>Показывать заряды</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pbAddCharge">
          <property name="text">
           <string>Добавить заряд</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pbRemoveAll">
          <property name="text">
           <string>Удалить всё</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="pbRenderP">
          <property name="text">
           <string>Построить потенциалы</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabConfig">
       <attribute name="title">
        <string>Настройки</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="gbFieldSize">
          <property name="title">
           <string>Размеры поля</string>
          </property>
          <layout class="QFormLayout" name="formLayout_2">
           <property name="fieldGrowthPolicy">
            <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
           </property>
           <item row="0" column="0">
            <widget class="QLabel" name="lbFieldWidth">
             <property name="text">
              <string>Ширина</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lbFieldHeight">
             <property name="text">
              <string>Высота</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="sbFieldWidth">
             <property name="minimum">
              <number>10</number>
             </property>
             <property name="maximum">
              <number>2000</number>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="sbFieldHeight">
             <property name="minimum">
              <number>10</number>
             </property>
             <property name="maximum">
              <number>2000</number>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="renderHints">
       <set>QPainter::Antialiasing|QPainter::TextAntialiasing</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pbAddCharge</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addCharge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>1</x>
     <y>122</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbRemoveAll</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>284</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>371</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbRemove</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeCharge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>89</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeChargeX(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>252</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeChargeY(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbCharge</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>changeChargeQ(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>215</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel">
     <x>442</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbRenderP</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>renderP()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>366</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbChargesVisible</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setChargesVisible(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbFieldWidth</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>fieldSizeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>58</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbFieldHeight</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>fieldSizeChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addCharge()</slot>
  <slot>removeCharge()</slot>
  <slot>removeAll()</slot>
  <slot>changeChargeX(double)</slot>
  <slot>changeChargeY(double)</slot>
  <slot>changeChargeQ(double)</slot>
  <slot>renderP()</slot>
  <slot>setChargesVisible(int)</slot>
  <slot>fieldSizeChanged()</slot>
 </slots>
</ui>
